### DISCUSSION TITLE
Would it make sense to support kaitai struct file definitions?

### DISCUSSION THREAD

**Discussion by mofosyne** [WerWolv/ImHex#1779] [General]
https://github.com/kaitai-io/kaitai_struct

It looks like katai struct is trying to create a universal file parser and they have their own language for defining file formats, maybe you could add support for it if (especially if it becomes more popular)

----

Example of a BMP bitmap image file format definition https://formats.kaitai.io/bmp/

**Comment by WerWolv** [comment]
Hey, you're far from the first one to ask for Kaitai support. My opinion on it is that Kaitai is great but I rather want to develop my own solution for it because I enjoy doing so. Converting scripts over from Kaitai is usually not super difficult though and generally they are more readable in the pattern language.
